### PR TITLE
[v0.9] update cometbft dep for security alert

### PR DESCRIPTION
**Pull request by jchappelow**
Resolves a dependabot alert regarding an unrecoverable cometbft panic.

https://github.com/kwilteam/kwil-db/pull/1097 for `release-v0.9`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,12 @@
 module github.com/kwilteam/kwil-db
 
-go 1.22
+go 1.22.7
+
+toolchain go1.23.3
```
Comment (jchappelow) [inline go.mod:5]: I have no clue why it's doing this and I can't seem to make it go away.

Comment (brennanjl) [reply go.mod:5]: Should we try to resolve or just let it be?

**Review by brennanjl — APPROVED** [review]
lgtm but will wait for you to merge if you want to try to figure out the toolchain thing. No opinion from my side

**Comment by jchappelow** [comment]
> lgtm but will wait for you to merge if you want to try to figure out the toolchain thing. No opinion from my side

OK.  I'm taking a look at docs, but it doesn't seem to affect much as long as the `go 1.22.7` line is there.

**Comment by jchappelow** [comment]
Ok, toolchain doesn't seem to be coming back now.  Hmm.  The comet update forces the 1.22**.7** change, but at least the toolchain assertion is gone

**Review by brennanjl — APPROVED** [review]

**Pull request merged**